dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/SecurityOwnershipValidator.java

```diff
@@ -98,15 +96,39 @@ public void validate(Reporter reporter, TrackerBundle bundle, Enrollment enrollm
       }
     }
 
-    checkWriteEnrollmentAccess(reporter, bundle, enrollment, program, ownerOrgUnit, trackedEntity);
+    checkWriteEnrollmentAccess(
+        reporter, bundle, enrollment, program, ownerOrgUnit, trackedEntity.getUid());
+  }
+
+  private TrackedEntity getTrackedEntityWhenStrategyCreate(
+      TrackerBundle bundle, Enrollment enrollment) {
+    TrackedEntity trackedEntity =
+        bundle.getPreheat().getTrackedEntity(enrollment.getTrackedEntity());
+
+    if (trackedEntity == null) {
+      Optional<org.hisp.dhis.tracker.imports.domain.TrackedEntity> te =
+          bundle.findTrackedEntityByUid(enrollment.getTrackedEntity());
+
+      return te.map(
+              entity -> {
+                TrackedEntity newEntity = new TrackedEntity();
+                newEntity.setUid(entity.getUid());
+                newEntity.setOrganisationUnit(
+                    bundle.getPreheat().getOrganisationUnit(entity.getOrgUnit()));
+                return newEntity;
+              })
+          .get();
```

Comment:
@enricocolasante is it fine to assume the tracked entity will always be present in the payload when not found in the database?

Reply:
Yes, we already checked that it exists in [MetaValidator](https://github.com/dhis2/dhis2-core/blob/5da7aa8724d65a3fda83c57afa7bba1962c3d66b/dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/MetaValidator.java#L55-L59)